kernel/src/engine/parquet_stats_skipping.rs

```diff
@@ -0,0 +1,1267 @@
+                    inverted
+                ),
+                Some(!inverted),
+                "{col} == {MID} (min: {MID}, max: {MID}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(LO.into(), HI.into()).apply_binary(
+                    Equal,
+                    col,
+                    &MID.into(),
+                    inverted
+                ),
+                Some(true), // min..max range includes both EQ and NE
+                "{col} == {MID} (min: {LO}, max: {HI}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(LO.into(), MID.into()).apply_binary(
+                    Equal,
+                    col,
+                    &HI.into(),
+                    inverted
+                ),
+                Some(inverted),
+                "{col} == {HI} (min: {LO}, max: {MID}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(MID.into(), HI.into()).apply_binary(
+                    Equal,
+                    col,
+                    &LO.into(),
+                    inverted
+                ),
+                Some(inverted),
+                "{col} == {LO} (min: {MID}, max: {HI}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(MID.into(), MID.into()).apply_binary(
+                    NotEqual,
+                    col,
+                    &MID.into(),
+                    inverted
+                ),
+                Some(inverted),
+                "{col} != {MID} (min: {MID}, max: {MID}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(LO.into(), HI.into()).apply_binary(
+                    NotEqual,
+                    col,
+                    &MID.into(),
+                    inverted
+                ),
+                Some(true), // min..max range includes both EQ and NE
+                "{col} != {MID} (min: {LO}, max: {HI}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(LO.into(), MID.into()).apply_binary(
+                    NotEqual,
+                    col,
+                    &HI.into(),
+                    inverted
+                ),
+                Some(!inverted),
+                "{col} != {HI} (min: {LO}, max: {MID}, inverted: {inverted})"
+            );
+
+            expect_eq!(
+                MinMaxTestFilter::new(MID.into(), HI.into()).apply_binary(
+                    NotEqual,
+                    col,
+                    &LO.into(),
+                    inverted
+                ),
+                Some(!inverted),
+                "{col} != {LO} (min: {MID}, max: {HI}, inverted: {inverted})"
+            );
+        }
+    }
+
+    #[test]
+    fn test_binary_lt_ge() {
+        use BinaryOperator::*;
+
+        const LO: Scalar = Scalar::Long(1);
```

Comment:
Can we use some other scalar types between say the eq_be tests and the lt_gt ones? Probably should be minimal change here but we can exercise the scalar stuff for more than just `Long`

Reply:
This test is only trying to verify the conversion regular predicate to data skipping predicate. The different scalar comparisons (by both matched and mismatched types) are already exercised thoroughly by `test_binary_scalars`. 

Is there a particular corner case you worry about, that would make the data skipping code misbehave based on the type of scalar involved?

Reply:
Update: Added a negative test, where the literal and column types mismatch.